src/pages/Task/components/MainContent/Comments/AddComment/AddComment.tsx

```diff
@@ -0,0 +1,136 @@
+  { icon: <AlternateEmailIcon className="sm-icon" /> },
+  { icon: <SentimentSatisfiedRoundedIcon className="sm-icon" /> },
+  { icon: <VideoLabelIcon className="sm-icon" /> },
+];
+
+interface IRouteParams {
+  boardId: string;
+  taskId: string;
+}
+
+const AddComment: React.FC = () => {
+  const [isWritingComment, setIsWritingComment] = useState(false);
+  const [newCommentText, setNewCommentText] = useState('');
+  const queryClient = useQueryClient();
+  const { user } = useAuth();
+  const { taskId, boardId } = useParams<IRouteParams>();
+
+  const taskData = queryClient.getQueryData<ITaskDetails>(['task', taskId])!;
+  const boardData = queryClient.getQueryData<IBoard>(['board', boardId])!;
+  const { owners, members } = boardData;
+  const users = [...owners, ...members];
+  const { comments, cover } = taskData;
+
+  const { mutate: addComment } = useMutation(
+    () => addOneComment(newCommentText, user, taskId, comments, cover, users),
+    taskMutationConfig(taskId, queryClient, {
+      key: taskMutationKeys.addComment,
+    })
+  );
+
+  const addCommentHandler = (event: React.FormEvent) => {
+    event.preventDefault();
+    if (newCommentText && newCommentText.length) {
+      const newComment: IComment = {
+        commentId: `optimistic${uuidv4()}`,
+        commentText: newCommentText,
+        createdAt: '',
+        updatedAt: '',
+        taskId,
+        author: user,
+      };
+      const newCommentsList = [newComment, ...taskData.comments];
+      addComment({ ...taskData, comments: newCommentsList });
+      setNewCommentText('');
+      setIsWritingComment(false);
+    }
+  };
+
+  const SaveButton = () => (
+    <StyledSaveButton
+      onClick={addCommentHandler}
+      disabled={newCommentText.length === 0}
+    >
+      Save
+    </StyledSaveButton>
+  );
+
+  return (
+    <StyledWrapper>
+      <StyledAvatar src={user.profileImg}>
+        {getAvatarFallbackName(user.username)}
+      </StyledAvatar>
+      <ClickAwayListener onClickAway={() => setIsWritingComment(false)}>
+        <StyledAccordion expanded={isWritingComment}>
+          <StyledAccordionSummary iswriting={+isWritingComment}>
+            <StyledTextField
+              onFocus={() => setIsWritingComment(true)}
+              multiline
+              variant="outlined"
+              placeholder="Write a comment..."
+              iswriting={+isWritingComment}
+              value={newCommentText}
+              onChange={(e) => setNewCommentText(e.target.value)}
+            />
+          </StyledAccordionSummary>
+          <StyledAccordionDetails>
+            {!newCommentText.length && (
+              <div className="not-allowed">
+                <SaveButton />
+              </div>
+            )}
+            {newCommentText.length > 0 && <SaveButton />}
+            <StyledIconBtnContainer>
+              {iconBtnList.map(({ icon }) => (
+                <StyledIconButton key={uuidv4()}>{icon}</StyledIconButton>
```

Comment:
Avoid using uuidv4 as react component keys. This should be done only when its really necessary, as the uuidv4() return different keys on every render and react mounts it again. For more see:
https://stackoverflow.com/questions/52705335/will-using-uuid-to-as-list-keys-cause-unnecessary-re-renders-in-react

In the iconBtnList array, you can simply add id key in the objects to avoid it here